### PR TITLE
[BugFix] Fixing CVRPTW bugs during multistart solution generation

### PR DESCRIPTION
**Pull request by ahottung**
## Description
This pull request fixes some very minor bugs that occur when using the CVRPTW env during multistart solution generation.  Additionally, it adds the CVRPTW to the problems considered in the policy test cases.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ x] Bug fix (non-breaking change which fixes an issue)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] My change requires a change to the documentation.
- [ x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: rl4co/envs/routing/cvrptw/env.py [hunk 1]
```diff
@@ -114,7 +110,11 @@ def get_action_mask(td: TensorDict) -> torch.Tensor:
         current_loc = gather_by_index(td["locs"], td["current_node"]).reshape(
             [batch_size, 2]
         )
-        dist = get_distance(current_loc, td["locs"].transpose(0, 1)).transpose(0, 1)
+        dist = (
+            get_distance(current_loc, td["locs"].transpose(0, 1))
```
Comment (fedebotu) [inline rl4co/envs/routing/cvrptw/env.py:114]: You can achieve the same as above with the `v2` which is a bit more readable and faster:

```python
from tensordict import TensorDict
import torch

from rl4co.utils.ops import gather_by_index, get_distance

td = TensorDict(
    {
        "locs": torch.rand(32, 10, 2),
        "current_node": torch.randint(0, 10, (32,1))
    },
    batch_size = 32
)

def get_distv1():
    batch_size = td["locs"].shape[0]
    current_loc = gather_by_index(td["locs"], td["current_node"]).reshape(
        [batch_size, 2]
    )
    dist = (
        get_distance(current_loc, td["locs"].transpose(0, 1))
        .transpose(0, 1)
        .contiguous()
    )
    return dist

def get_distv2():
    cur_loc = gather_by_index(td["locs"], td["current_node"])
    d_ij = get_distance(cur_loc[..., None, :], td["locs"])
    return d_ij

torch.allclose(get_distv1(), get_distv2())
```

**Review by fedebotu — APPROVED** [review]

**Comment by fedebotu** [comment]
LGTM, @cbhua @ngastzepeda you can merge after your approval

**Review by cbhua — APPROVED** [review]
Really good! Thanks for the debugging!

**Review by ngastzepeda — COMMENTED** [review]
I'd suggest to set `squeeze=False` for all `gather_by_index()` occurrences to prevent issues with `batch_size=1`.

File: rl4co/envs/routing/cvrptw/env.py [hunk 2]
```diff
@@ -1,21 +1,17 @@
...
 
-from rl4co.envs.routing.cvrp.env import CVRPEnv
-from rl4co.envs.routing.cvrp.generator import CAPACITIES
-from rl4co.utils.ops import gather_by_index, get_distance
 from rl4co.data.utils import (
     load_npz_to_tensordict,
     load_solomon_instance,
     load_solomon_solution,
 )
+from rl4co.envs.routing.cvrp.env import CVRPEnv
```
[the remark this answers is not in the document]
Comment (ngastzepeda) [reply rl4co/envs/routing/cvrptw/env.py:13]: I seem to be the only one with this issue, but in general: should we include the generators in the `__init__` in the routing folder? I went ahead and added them, if you had left them out on purpose, please feel free to remove them again :sweat_smile:

File: rl4co/envs/routing/cvrptw/env.py [hunk 3]
```diff
@@ -110,11 +106,8 @@ def get_action_mask(td: TensorDict) -> torch.Tensor:
         The vehicle can only visit a location if it can reach it in time, i.e. before its time window ends.
         """
         not_masked = CVRPEnv.get_action_mask(td)
-        batch_size = td["locs"].shape[0]
-        current_loc = gather_by_index(td["locs"], td["current_node"]).reshape(
-            [batch_size, 2]
-        )
-        dist = get_distance(current_loc, td["locs"].transpose(0, 1)).transpose(0, 1)
+        current_loc = gather_by_index(td["locs"], td["current_node"])
```
[the remark this answers is not in the document]
Comment (fedebotu) [reply rl4co/envs/routing/cvrptw/env.py:109]: Right, I agree with you!

File: rl4co/envs/routing/cvrptw/env.py [hunk 4]
```diff
@@ -1,21 +1,17 @@
...
+
 import torch
+
 from tensordict.tensordict import TensorDict
-from torchrl.data import (
-    BoundedTensorSpec,
-    CompositeSpec,
-    UnboundedContinuousTensorSpec,
-)
+from torchrl.data import BoundedTensorSpec, CompositeSpec, UnboundedContinuousTensorSpec
```
[the remark this answers is not in the document]
Comment (fedebotu) [reply rl4co/envs/routing/cvrptw/env.py:6]: We use Black + Ruff as described [here](https://github.com/ai4co/rl4co/blob/c546234360e4a5e772aa8a3ae8e58e85d0698492/pyproject.toml#L100)

File: rl4co/envs/routing/cvrptw/env.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ngastzepeda) [reply rl4co/envs/routing/cvrptw/env.py:13]: Providing them in the `__init__` would also make it easier for the users to call the generator themselves and fill it with parameters, as they're supposed to pass the generator as a parameter to their env, anyways :)

**Comment by fedebotu** [comment]
LGTM, merging now!

**Pull request merged**